src/spi_bus.cpp

```diff
@@ -0,0 +1,364 @@
+            break;
+        }
+    }
+    if(controls_have_changed)
+    {
+        send_newest_controls();
+    }
+}
+
+void SpiBus::send_newest_command()
+{
+    for(size_t i = 0 ; i < nb_udrivers_ ; ++i)
+    {
+        // Some aliases
+        MotorDriver& udriver = main_board_interface_->motor_drivers[i];
+        const uint32_t& command_id = command_[i]->newest_element().id_;
+        const int32_t& command_content = command_[i]->newest_element().content_;
+        switch (command_id)
+        {
+            case MotorBoardCommand::ENABLE_SYS:
+                command_content == MotorBoardCommand::ENABLE ?
+                    udriver.Enable() : udriver.Disable();
+                break;
+            case MotorBoardCommand::ENABLE_MTR1:
+                command_content == MotorBoardCommand::ENABLE ?
+                    udriver.motor1->Enable() : udriver.motor1->Disable();
+                break;
+            case MotorBoardCommand::ENABLE_MTR2:
+                command_content == MotorBoardCommand::ENABLE ?
+                    udriver.motor2->Enable() : udriver.motor2->Disable();
+                break;
+            case MotorBoardCommand::ENABLE_VSPRING1:
+                break;
+            case MotorBoardCommand::ENABLE_VSPRING2:
+                break;
+            case MotorBoardCommand::SEND_CURRENT:
+                break;
+            case MotorBoardCommand::SEND_POSITION:
+                break;
+            case MotorBoardCommand::SEND_VELOCITY:
+                break;
+            case MotorBoardCommand::SEND_ADC6:
+                break;
+            case MotorBoardCommand::SEND_ENC_INDEX:
+                break;
+            case MotorBoardCommand::SEND_ALL:
+                break;
+            case MotorBoardCommand::SET_CAN_RECV_TIMEOUT:
+                break;
+            case MotorBoardCommand::ENABLE_POS_ROLLOVER_ERROR:
+                command_content == MotorBoardCommand::ENABLE ?
+                    udriver.EnablePositionRolloverError() :
+                    udriver.DisablePositionRolloverError();
+                break;        
+            default:
+                throw std::runtime_error("SpiBus::send_if_input_changed:"
+                                        " Error cannot send unkown command");
+                break;
+        } // end switch
+    } // end for
+}
+
+void SpiBus::send_newest_controls()
+{
+    // send controls -----------------------------------------------------------
+    for(size_t i = 0; i < control_.size(); ++i)
+    {
+        if(control_[i]->length() == 0)
+        {
+            rt_printf("You tried to send control on the motor[%ld] "
+                      "but no control has been set\n", i);
+            exit(-1);
+        }
+        MotorInterface::ScalarTimeseries::Index timeindex =
+            control_[i]->newest_timeindex();
+
+        double control = (*control_[i])[timeindex];
+        
+        main_board_interface_->motors[i].SetCurrentReference(control);
```

Comment:
are you 100 sure that main_board_interface_ uses the same indexing of the motors as we do in this class? just to make sure..